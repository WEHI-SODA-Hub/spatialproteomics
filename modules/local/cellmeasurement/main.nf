process CELLMEASUREMENT {
    tag "$meta.id"
    label 'process_medium_memory'

    conda "${moduleDir}/environment.yml"
    container "ghcr.io/wehi-soda-hub/cellmeasurement:latest"

    input:
    tuple val(meta),
        path(tiff),
        path(nuclear_mask),
        path(whole_cell_mask),
        val(cell_expansion),
        val(skip_measurements)
    val(pixel_size_microns)

    output:
    tuple val(meta), path("*.geojson"), emit: annotations
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    def cell_expansion_arg = cell_expansion.first() != [] ? "--cell-expansion ${cell_expansion.first()}" : ''
    def skip_measurements_arg = skip_measurements ? '--skip-measurements=true' : '--skip-measurements=false'
    """
    /entrypoint.sh \\
        --args="${args} \\
            --nuclear-mask=${nuclear_mask} \\
            --whole-cell-mask=${whole_cell_mask} \\
            --tiff-file=${tiff} \\
            --output-file=${prefix}.geojson \\
            ${cell_expansion_arg} \\
            ${skip_measurements_arg}"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        cellmeasurement: \$(/entrypoint.sh --version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.geojson

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        cellmeasurement: \$(/entrypoint.sh --version)
    END_VERSIONS
    """
}
