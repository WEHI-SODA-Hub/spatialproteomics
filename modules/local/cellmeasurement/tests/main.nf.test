nextflow_process {

    name "Test Process CELLMEASUREMENT"
    script "../main.nf"
    process "CELLMEASUREMENT"

    tag "modules"
    tag "modules_nfcore"
    tag "cellmeasurement"

    test("cellmeasurement - basic functionality") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file('${projectDir}/tests/data/mesmer/test_data.tiff'),
                    file('${projectDir}/tests/data/mesmer/test_data_nuclear.tiff'),
                    file('${projectDir}/tests/data/mesmer/test_data_whole-cell.tiff'),
                    [[]], // cell_expansion - empty list
                    [false] // skip_measurements
                ]
                input[1] = 0.5 // pixel_size_microns
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("cellmeasurement - with cell expansion and skip measurements") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test_expansion' ], // meta map
                    file('${projectDir}/tests/data/mesmer/test_data.tiff'),
                    file('${projectDir}/tests/data/mesmer/test_data_nuclear.tiff'),
                    file('${projectDir}/tests/data/mesmer/test_data_whole-cell.tiff'),
                    [2.0], // cell_expansion - with value
                    [true] // skip_measurements
                ]
                input[1] = 0.5 // pixel_size_microns
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("cellmeasurement - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test_stub' ], // meta map
                    file('${projectDir}/tests/data/mesmer/test_data.tiff'),
                    file('${projectDir}/tests/data/mesmer/test_data_nuclear.tiff'),
                    file('${projectDir}/tests/data/mesmer/test_data_whole-cell.tiff'),
                    [], // cell_expansion - empty list
                    false // skip_measurements
                ]
                input[1] = 0.5 // pixel_size_microns
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
