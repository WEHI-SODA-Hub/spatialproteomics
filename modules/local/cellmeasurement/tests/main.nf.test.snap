{
    "cellmeasurement - basic functionality": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": [
                                "analyse_samplesheet.csv:md5,bf995bff1feac8b4fc2b4f46e7ae5b08"
                            ]
                        },
                        "test.geojson:md5,775b8fe1e8bd8252b390691fd0fefff4"
                    ]
                ],
                "1": [
                    "versions.yml:md5,911b2db92c964303bac36ad611f832a8"
                ],
                "annotations": [
                    [
                        {
                            "id": [
                                "analyse_samplesheet.csv:md5,bf995bff1feac8b4fc2b4f46e7ae5b08"
                            ]
                        },
                        "test.geojson:md5,775b8fe1e8bd8252b390691fd0fefff4"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,911b2db92c964303bac36ad611f832a8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-07-09T11:20:58.086962155"
    },
    "cellmeasurement - with cell expansion and skip measurements": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_expansion"
                        },
                        "test_expansion.geojson:md5,d0d6433dbc4118045bb0111d4e66b816"
                    ]
                ],
                "1": [
                    "versions.yml:md5,911b2db92c964303bac36ad611f832a8"
                ],
                "annotations": [
                    [
                        {
                            "id": "test_expansion"
                        },
                        "test_expansion.geojson:md5,d0d6433dbc4118045bb0111d4e66b816"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,911b2db92c964303bac36ad611f832a8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-07-02T16:14:05.984289374"
    },
    "cellmeasurement - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_stub"
                        },
                        "test_stub.geojson:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,911b2db92c964303bac36ad611f832a8"
                ],
                "annotations": [
                    [
                        {
                            "id": "test_stub"
                        },
                        "test_stub.geojson:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,911b2db92c964303bac36ad611f832a8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-07-02T13:00:16.806661896"
    },
    "cellmeasurement - skip measurements": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_expansion"
                        },
                        "test_expansion.geojson:md5,5517147da4081e48cc2c5d6761c5d48a"
                    ]
                ],
                "1": [
                    "versions.yml:md5,911b2db92c964303bac36ad611f832a8"
                ],
                "annotations": [
                    [
                        {
                            "id": "test_expansion"
                        },
                        "test_expansion.geojson:md5,5517147da4081e48cc2c5d6761c5d48a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,911b2db92c964303bac36ad611f832a8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-07-09T11:21:45.769311282"
    }
}